daemon/cmd/daemon.go

```diff
@@ -506,7 +506,7 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 	}
 
 	if option.Config.EnableIPv4EgressGateway {
-		d.egressGatewayManager = egressgateway.NewEgressGatewayManager(&d)
+		d.egressGatewayManager = egressgateway.NewEgressGatewayManager(&d, d.identityAllocator)
```

Comment:
A bit curious, is this identityAllocator local to the node? Or does it also store labels for identities created on remote nodes?

Reply:
It's a caching allocator that has both global and local identities. Details:
https://github.com/cilium/cilium/blob/master/pkg/identity/cache/allocator.go#L69
When looking up, it prefers the local cache.

Reply:
it says it is from the kvstore: https://github.com/cilium/cilium/blob/master/pkg/identity/cache/allocator.go#L71 which got me thinking is the "global" cache used when kvstore is CRD mode?

Reply:
Yes i think it uses both global and local cache. For example when looking by ID, here it looks through the local identities first: https://github.com/cilium/cilium/blob/82c742f2e9fb65fb3fc392c7dcb3b4e22b69c650/pkg/identity/cache/cache.go#L281

Reply:
I see, makes sense, thanks for the pointers!